### PR TITLE
api/tes: ListTasks state filter

### PR DESCRIPTION
**Pull request by buchanae**
Taking a swing at ListTasks filtering. Intended to be quick and dirty to get this ball rolling, will refine towards the end of this PR.

### CODE REVIEW

**Comment by adamstruck** [comment]
Nice work.  I have two comments:

1) Can we change `state_filter` in the `ListTasksRequest` to `state`
2) Add a CLI flag to `funnel task list`

**Comment by buchanae** [comment]
Ok, address the feedback and added a filter to the webdash UI. The UI style is pretty ugly, I'll revisit that in a future PR.

File: cmd/task/list.go [hunk 1]
```diff
@@ -10,10 +10,9 @@ import (
 )
 
 // List runs the "task list" CLI command, which connects to the server,
-// calls ListTasks() and requests the given task view. Results may be filtered
-// client-side using the "taskState" and "taskName" args. Output is written
-// to the given writer.
-func List(server, taskView, pageToken string, pageSize uint32, all bool, writer io.Writer) error {
+// calls ListTasks() and requests the given task view.
+// Output is written to the given writer.
+func List(server, taskView, pageToken, stateFilter string, pageSize uint32, all bool, writer io.Writer) error {
```
Comment (adamstruck) [inline cmd/task/list.go]: `stateFilter` is never used.

**Comment by adamstruck** [comment]
For the webdash, it would be nice if the url was updated as filters are applied.

File: Makefile [hunk 2]
```diff
@@ -95,59 +94,59 @@ start-elasticsearch:
...
+	@go test -timeout 120s ./tests/gridengine -funnel-config `pwd`/tests/gridengine.config.yml
 
 test-pbs-torque:
 	@docker pull ohsucompbio/pbs-torque
-	@go test -timeout 120s ./tests/pbs -funnel-config $(CONFIGDIR)/pbs.config.yml
+	@go test -timeout 120s ./tests/pbs -funnel-config `pwd`/tests/pbs.config.yml
 
 test-amazon-s3:
-	@go test ./tests/storage -funnel-config $(CONFIGDIR)/s3.config.yml -run TestAmazonS3
+	@go test ./tests/storage -funnel-config `pwd`/s3.config.yml -run TestAmazonS3
```
Comment (adamstruck) [inline Makefile]: Here and several places below: 
```
`pwd`/ -> `pwd`/tests
```

**Comment by buchanae** [comment]
> For the webdash, it would be nice if the url was updated as filters are applied.

I agree. Just for the sake of time, I'm going to punt on this for now though.

https://github.com/ohsu-comp-bio/funnel/issues/473

File: cmd/task/list.go [hunk 3]
```diff
@@ -26,12 +25,20 @@ func List(server, taskView, pageToken string, pageSize uint32, all bool, writer
 
 	output := &tes.ListTasksResponse{}
 
+	state := tes.Unknown
+	if stateInt, ok := tes.State_value[stateFilter]; ok {
```
Comment (adamstruck) [inline cmd/task/list.go]: The logic here should probably be a closer match to the `getTaskView` utility above so we get case insensitivity and an informative error message.

**Comment by buchanae** [comment]
This is ready to go. I've restarted the TravisCI build multiple times.

**Pull request merged**